### PR TITLE
S3 PART 1 - Write to S3

### PR DESCRIPTION
**Pull request by colegottdank**
- Writes to S3 in the worker
- To do that, it returns the body separately in places where we are about to insert it into the postgresDB
- Adds python script to run Minio locally

### CODE REVIEW

File: worker/src/db/S3Client.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  async store(url: string, value: string): Promise<Result<string, string>> {
+    const signedRequest = await this.awsClient.sign(url, {
+      method: "PUT",
+      body: value,
+      headers: {
+        "Content-Type": "application/json",
+      },
+    });
+
+    const response = await fetch(signedRequest.url, signedRequest);
```
Comment (chitalian) [inline worker/src/db/S3Client.ts:53]: `fetch` is so annoying. I was running this and without the url service being up I was getting 

```
✘ [ERROR] Uncaught (in promise) Error: Network connection lost.
```

This still allowed the logs to go through...

So log in clickhouse and webhooks were never getting hit.

Here is the test I ran... 
```
    console.log("test point 1");
    const s3Result = await db.s3Client.storeRequestResponse(
      authParams.organizationId,
      this.request.requestId,
      requestResult.data.body,
      responseResult.data.body
    );
    console.log("test point 2");
```


So basically I think we need to wrap the whole store in a try catch

File: worker/src/lib/dbLogger/DBLoggable.ts [hunk 2]
```diff
@@ -605,12 +634,42 @@ export class DBLoggable {
...
+      authParams.organizationId,
+      this.request.requestId,
+      requestResult.data.body,
+      responseResult.data.body
+    );
+
+    if (s3Result.error) {
+      console.log("Error storing request response", s3Result.error);
+      return err(s3Result.error);
+    }
```
Comment (chitalian) [inline worker/src/lib/dbLogger/DBLoggable.ts:668]: Also I dont think we should return here. Just incase

Comment (chitalian) [reply worker/src/lib/dbLogger/DBLoggable.ts:668]: for now...

File: README.md [hunk 3]
```diff
@@ -93,6 +93,21 @@ brew install supabase/tap/supabase
 python3 clickhouse/ch_hcone.py --start
 ```
 
+## Install and setup MinIO
+
+```bash
+# Install minio
+pip install minio
```
Comment (chitalian) [inline README.md:100]:
```suggestion
python3 -m pip install minio
```

File: worker/src/lib/dbLogger/DBLoggable.ts (hunk 2 above)
Comment (colegottdank) [reply worker/src/lib/dbLogger/DBLoggable.ts]: updated

File: worker/src/db/S3Client.ts (hunk 1 above)
Comment (colegottdank) [reply worker/src/db/S3Client.ts]: added try catch

**Pull request merged**